include/drivers/i2c.h

```diff
@@ -337,6 +337,151 @@ struct i2c_slave_config {
 	const struct i2c_slave_callbacks *callbacks;
 };
 
+#ifdef CONFIG_I2C_STATS
+
+#include <stats/stats.h>
+
+STATS_SECT_START(i2c)
+STATS_SECT_ENTRY32(bytes_read)
+STATS_SECT_ENTRY32(bytes_written)
+STATS_SECT_ENTRY32(message_count)
+STATS_SECT_ENTRY32(transfer_call_count)
+STATS_SECT_END;
+
+STATS_NAME_START(i2c)
```

Comment:
this seems to expand to a definition:
```
#define STATS_NAME_START(sectname__) \
	const struct stats_name_map STATS_NAME_MAP_NAME(sectname__)[] = {
```
this will get pasted to every file including `drivers/i2c.h`, right?

Reply:
It expands to

``` c
static const struct stats_name_map STATS_NAME_MAP_NAME(sectname__)[] = {   
```
This could be done using an extern const if preferred, but then the stats struct and name will be split across two files which, in my opinion, will make it slightly more involved when adding members as now you need to remember to update two files instead of the one header as it is here. It also requires an compilation unit (ex i2c_common.c) to always exist for each driver class which may/may not exist yet for all of them. I wanted to make this incredibly easy to replicate across device classes without a lot of extra work beyond copy and paste this file with some minor modifications for each device class. I also wanted to make it easy to look and see any obviously missing/misnamed member name mappings.